src/main/java/ailurus/Storage.java

```diff
@@ -0,0 +1,93 @@
+package ailurus;
+
+import ailurus.task.Task;
+import ailurus.task.TaskList;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private String directoryName;
+    private String filename;
+    private File directory = null;
+    private File ailurusObj = null;
+    private FileWriter writer = null;
+
+    /**
+     * Constructor for storage
+     *
+     * @param directoryName name of directory to be created and for storage of data
+     * @param filename name of file to be created and for storage of data
+     */
+    public Storage(String directoryName, String filename) {
+        this.directoryName = directoryName;
+        this.filename = filename;
+        this.directory = new File(directoryName);
+        this.ailurusObj = new File(directoryName.concat(filename));
+
+        if (!directory.exists()) {
+            // create new directory
+            directory.mkdir();
+        }
+
+        if (!ailurusObj.exists()) {
+            // create new file
+            try {
+                ailurusObj.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+
+    /**
+     * Reads and loads the tasks in the data file, and creates a FileWriter
+     * that must be followed with unload() to close writer
+     *
+     * @return list of tasks that is loaded from data file
+     */
+    public ArrayList<Task> load() throws AilurusException {
+        ArrayList<Task> list = new ArrayList<>();
+        // load tasks from file
+        try {
+            Scanner reader = new Scanner(ailurusObj);
+            while (reader.hasNextLine()) {
+                Task task = Parser.parseData(reader.nextLine());
+                if (task != null) {
+                    list.add(task);
+                }
+            }
+            this.writer = new FileWriter(this.directoryName.concat(this.filename));
+        } catch (FileNotFoundException e) {
+            throw new AilurusException(AilurusException.Error.LOAD);
+        } catch (IOException e) {
+            throw new AilurusException(AilurusException.Error.LOAD);
+        }
+        return list;
```

Comment:
use (FileNotFoundException e |  IOException e) could be better?

Reply:
Tried that and this is the prompt: Types in multi-catch must be disjoint: 'java.io.FileNotFoundException' is a subclass of 'java.io.IOException. So I believe I can just catch for IOException. Thank you!